ray-operator/controllers/ray/common/pod.go

```diff
@@ -368,6 +368,15 @@ func BuildPod(ctx context.Context, podTemplateSpec corev1.PodTemplateSpec, rayNo
 		cmd += convertCmdToString(pod.Spec.Containers[utils.RayContainerIndex].Args)
 	}
 
+	if rayNodeType == rayv1.HeadNode && gcsOptions != nil {
+		// If gcsOptions.RedisPassword is present, it will be put into env vars later.
+		// Here, we add --redis-password flag to the ray start command with value "$REDIS_PASSWORD".
+		// $REDIS_PASSWORD will be replaced with the env var by the shell.
+		if gcsOptions.RedisPassword != nil {
+			rayStartParams["redis-password"] = "$REDIS_PASSWORD"
+		}
```

Comment:
If `gcsOptions.RedisPassword` is present, add `--redis-password` flags to the ray start command.

`$REDIS_PASSWORD` will be replaced with the env var by the shell.